### DISCUSSION TITLE
Zustand file structure or Pattern

### DISCUSSION THREAD

**Discussion by umararshad4** [pmndrs/zustand#1259] [General]
what file structure or pattern to follow with zustand using react.js

**Comment by lkdm** [comment]
Zustand is unopinionated. If you're struggling, like I was, I recommend reading the [new docs](https://github.com/pmndrs/zustand/tree/main/docs/guides) to get a vague idea of different ways you could structure your store.

Some of the options you have are:

- Do you use Zustand [the default way](https://github.com/pmndrs/zustand/blob/main/readme.md)?
- Do you put your actions in your store, or [have them separate](https://github.com/pmndrs/zustand/blob/main/docs/guides/practice-with-no-store-actions.md)?
- Do you [break your store into different slices](https://github.com/pmndrs/zustand/blob/main/docs/guides/flux-inspired-practice.md)?
- Do you deeply nest your state or normalise/flatten it? [See this article from Redux on normalising state](https://redux.js.org/usage/structuring-reducers/normalizing-state-shape).

As for file structure, I just put my store either in `hooks` directory or `store` depending on the size of the project.
